python/ray/air/_internal/tensorflow_utils.py

```diff
@@ -1,14 +1,9 @@
-from typing import TYPE_CHECKING, Dict, List, Optional, Tuple, Union
+from typing import Dict, Optional, Union
 
 import numpy as np
-import pyarrow
 import tensorflow as tf
 
 from ray.air.util.data_batch_conversion import _unwrap_ndarray_object_type_if_needed
-from ray.air.util.tensor_extensions.arrow import get_arrow_extension_tensor_types
-
-if TYPE_CHECKING:
-    from ray.data._internal.pandas_block import PandasBlockSchema
 
 
 def convert_ndarray_to_tf_tensor(
```

Comment:
This function is now duplicated across `ray.air` and `ray.data`, but its not dead code. 

In a follow-up, I think we should remove this and update the reference in `tensorflow_predictor` to use the one defined in `ray.data`.